Automattic/wp-calypso
# Update backups date navigation to add a date picker component

#### Changes proposed in this Pull Request

* This PR adds a new Date Picker button to the date controls on the Jetpack Backups page in Calypso.
* It introduces a new component to handle this button and removes the `< DateRangeSelector />` component that is in use currently on Jetpack Cloud to select dates for backups.

#### Testing instructions
* Apply this branch to your local Calypso environment
* Select a Jetpack site that has backups from the site selector
* Navigate to Jetpack > Backups in the left-hand menu to load the backups page for the site
* You should see the updated date selection bar, including a new date selection button on the right.
![Screen Shot 2021-09-17 at 5 36 28 PM](https://user-images.githubusercontent.com/18016357/133856140-8d91f450-ae72-47e7-9eee-f47b7c5bae4d.png)

* Clicking on the "Select Date" button should open a date-picker pop-over.
![Screen Shot 2021-10-04 at 5 25 44 PM](https://user-images.githubusercontent.com/18016357/135927767-2bcd7bab-1f4f-44f2-894d-d3e03fdb3730.png)

* Each date that should have a backup will display a small dot below the day number.
* Selecting a date from the pop-over modal should update the current backup showing on the page. 
* Note that the current, next and previous dates in the date selection navigation should also be updated after selecting a date from the date-picker.
* Open the date-picker again - the currently selected date should show highlighted with the Jetpack green accent color. Today's date should be marked with a dark grey circle.
* Check that dates in the future are not selectable within the date picker.
* Go back to previous months in the date picker, confirm that dates before the very first backup date for the site are not selectable. If the site's backup plan has a limit for how far back in time backups are available, the calendar should not show backups beyond that limit. This would be good to test/ simulate specifically. 
* If a site has days that do not have any successful backups (even if they are in the date range between the oldest backup and now) then those days should not be selectable on the calendar. This may be harder to simulate, but has some unit testing coverage.
* This functionality should show in both Calypso and Jetpack Cloud environments.

* Unit tests should pass for:
* `yarn run test-client client/my-sites/backup/backup-date-picker `
* `yarn run test-client client/my-sites/backup/status` 

#### Checking for new events
* With the backups page open, open the Network tab of your browser's developer tools
* Filter requests by `t.gif`
* You should see an event fire when the date picker is opened with the slug of `calypso_jetpack_backup_date_picker_open`
* When a valid date is selected and event should fire with the slug of `calypso_jetpack_backup_date_calendar`

#### Checking that everything still works without the new feature enabled
* At this time, the new calendar date picker will only be enabled in development and staging environments. Since there are some changes to the date selection components in general aside from the new date picker, it's also important to test with the feature disabled as well.
* In Calypso, to test functionality without the new date picker, use the `flags` query arg to disable the feature temporarily: `?flags=-jetpack/backups-date-calendar`. Make sure that the date selection functionality matches what is currently available in Calypso on wordpress.com.
* In your local Jetpack Cloud environment, use `?flags=jetpack/backups-date-picker,-jetpack/backups-date-calendar` to disable the new date picker and enable the current date selection feature that is available for Jetpack Cloud. Test and confirm that the date selection controls for backup match those available on cloud.jetpack.com and function as expected.
